botocore/serialize.py

```diff
@@ -1048,16 +1110,22 @@ def _serialize_type_map(self, xmlnode, params, shape, name):
         #       <value>val1</value>
         #     </entry>
         #  </MyMap>
-        node = ElementTree.SubElement(xmlnode, name)
-        # TODO: handle flattened maps.
```

Comment:
Have we tested against services to check that suddenly flattening these maps won't be a breaking change?  Or is that part of the spec?

Reply:
From discussion: There aren't any rest-xml services that use flattened maps. This is new functionality that shouldn't affect anybody on release, and therefore there isn't any risk of breaking customers relying on existing functionality.

Reply:
Script used to test:

```python
import botocore.session

session = botocore.session.get_session()
loader = session.get_component("data_loader")
services = loader.list_available_services("service-2")

for service in services:
    service_model = session.get_service_model(service)
    protocol = service_model.protocol
    
    # protocol check
    if protocol not in ['rest-xml']:
        continue
    
    for shape_name in service_model.shape_names:
        shape = service_model.shape_for(shape_name)
        if shape.serialization.get("flattened") and shape.type_name == "map":
            print(service, shape_name)
```